controllers/perses/deployment_controller.go

```diff
@@ -87,13 +92,28 @@ func (r *PersesReconciler) reconcileDeployment(ctx context.Context, req ctrl.Req
 			return subreconciler.RequeueWithError(err)
 		}
 
-		return subreconciler.RequeueWithDelay(time.Minute)
-	} else if err != nil {
-		dlog.WithError(err).Error("Failed to get Deployment")
+		return subreconciler.ContinueReconciling()
+	}
+
+	dep, err := r.createPersesDeployment(perses)
+	if err != nil {
+		dlog.WithError(err).Error("Failed to define new Deployment resource for perses")
+		return subreconciler.RequeueWithError(err)
+	}
 
+	// call update with dry run to fill out fields that are also returned via the k8s api
+	if err = r.Update(ctx, dep, client.DryRunAll); err != nil {
+		dlog.Error(err, "Failed to update Deployment with dry run")
 		return subreconciler.RequeueWithError(err)
 	}
 
+	if !reflect.DeepEqual(found, dep) {
```

Comment:
should we use `equality.Semantic.DeepEqual` from `k8s.io/apimachinery/pkg/api/equality` and include only the spec and/or other more granular fields that might actually re trigger a reconciliation?
```suggestion
	if !reflect.DeepEqual(found.Spec, dep.Spec) {
```

Reply:
I started with that and then changed it in case we wanted to redeploy resources based on the metadata, but I'm fine with that.